ui/calendar.js

```diff
@@ -135,16 +132,26 @@ return $.widget( "ui.calendar", {
 			return;
 		}
 
-		if ( this.date.month() !== oldMonth || this.date.year() !== oldYear ) {
-			this.refresh();
+		if ( this._needsRefresh() ) {
+			this._refresh();
 			this.grid.focus();
 		}
 
 		this._setActiveDescendant();
 	},
 
+	_needsRefresh: function() {
+		if ( this.date.month() !== this.viewDate.month() || this.date.year() !== this.viewDate.year() ) {
+			return !this.grid.find(
```

Comment:
Not sure what is meant here. The find() part is needed to make sure we do not have the day available due to eachDay changes (like in other-months demo).